File: app/src/main/java/com/zacharee1/systemuituner/activites/MainActivity.java
package com.zacharee1.systemuituner.activites;

import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.zacharee1.systemuituner.R;
import com.zacharee1.systemuituner.util.BillingUtil;
import com.zacharee1.systemuituner.misc.OptionSelected;
import com.zacharee1.systemuituner.handlers.RecreateHandler;
import com.zacharee1.systemuituner.util.Utils;

@SuppressWarnings("unused")
public class MainActivity extends AppCompatActivity
{
    private BillingUtil mBilling;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

//        if (sharedPreferences.getBoolean("use_fabric", true))
//            Fabric.with(this, new Crashlytics());

        setTheme(Utils.isInDarkMode(this) ? R.style.AppTheme_Dark : R.style.AppTheme);

        RecreateHandler.onCreate(this);

        setContentView(R.layout.activity_main);

        mBilling = new BillingUtil(this);
    }

    @Override
    protected void onDestroy()
    {
        RecreateHandler.onDestroy(this);
        super.onDestroy();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu)
    {
        getMenuInflater().inflate(R.menu.menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item)
    {
        return OptionSelected.doAction(item, this);
    }

    public void launchList(@Nullable View v) {
        startActivity(new Intent(this, ItemListActivity.class));
    }

    public void onDonatePayPalClicked(View v) {
        BillingUtil.onDonatePayPalClicked(this);
    }

    public void onDonate1Clicked(View v) {
        mBilling.onDonateClicked("donate_1");
    }

    public void onDonate2Clicked(View v) {
        mBilling.onDonateClicked("donate_2");
    }

    public void onDonate5Clicked(View v) {
        mBilling.onDonateClicked("donate_5");
    }

    public void onDonate10Clicked(View v) {
        mBilling.onDonateClicked("donate_10");
    }
}

File: app/src/main/java/com/zacharee1/systemuituner/activites/MainActivity.kt
package com.zacharee1.systemuituner.activites

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.zacharee1.systemuituner.R
import com.zacharee1.systemuituner.misc.OptionSelected
import com.zacharee1.systemuituner.util.BillingUtil
import com.zacharee1.systemuituner.util.prefs
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.layout_donate.*

class MainActivity : BaseAnimActivity() {
    private val billing by lazy { BillingUtil(this) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        open_tweaks.setOnClickListener { launchList() }
        paypal_button.setOnClickListener { onDonatePayPalClicked() }
        donate_1.setOnClickListener { onDonate1Clicked() }
        donate_2.setOnClickListener { onDonate2Clicked() }
        donate_5.setOnClickListener { onDonate5Clicked() }
        donate_10.setOnClickListener { onDonate10Clicked() }
    }

    override fun onResume() {
        super.onResume()

        if (prefs.hideWelcomeScreen) {
            launchList()
            finish()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return OptionSelected.doAction(item.itemId, this)
    }

    private fun launchList() {
        startActivity(Intent(this, OptionsActivity::class.java))
    }

    private fun onDonatePayPalClicked() {
        BillingUtil.onDonatePayPalClicked(this)
    }

    private fun onDonate1Clicked() {
        billing.onDonateClicked("donate_1")
    }

    private fun onDonate2Clicked() {
        billing.onDonateClicked("donate_2")
    }

    private fun onDonate5Clicked() {
        billing.onDonateClicked("donate_5")
    }

    private fun onDonate10Clicked() {
        billing.onDonateClicked("donate_10")
    }
}
